### PR TITLE
docs(helpers) add docs for function `wait_timer`

### PR DESCRIPTION
**Pull request by ADD-SP**
截屏2022-07-22 14 16 22

### CODE REVIEW

**Review by Tieske — CHANGES_REQUESTED** [review]
This doesn't even show up in the documentation? Can you try and run `ldoc` and verify?

File: spec/helpers.lua [hunk 1]
```diff
@@ -1388,13 +1388,17 @@ local function wait_until(f, timeout, step)
...
+--- 
+--- NOTE: this is a regular Lua function, not a Luassert assertion.
+---@function wait_timer
 ---@param timer_name_pattern string
 ---@param plain boolean
 ---@param mode string all-finish | all-running | any-finish | any-running | worker-wide-all-finish
 ---@param timeout? number optional, maximum time to wait (default = 2)
 ---@param admin_client_timeout? number optional, to override the default timeout setting
 ---@param forced_admin_port? number optional, to override the default port of admin API
+---@usage helpers.wait_timer("rate-limiting", true, "all-finish", 10)
```
Comment (Tieske) [inline spec/helpers.lua:1401]:
```suggestion
--- Wait for some timers, throws an error on timeout.
-- ?? what does it do? describe the "mode" parameter
-- @function wait_timer
-- @tparam string timer_name_pattern the call will apply to all timers matching
-- this string
-- @tparam boolean plain if truthy, the `timer_name_pattern` will be matched
-- plain, so without pattern matching
-- @tparam string mode one of: "all-finish", "all-running", "any-finish",
-- "any-running", or "worker-wide-all-finish"
-- @tparam[opt=2] number timeout maximum time to wait
-- @tparam[opt] number admin_client_timeout to override the default timeout setting
-- @tparam[opt] number forced_admin_port to override the default port of admin API
-- @return ????
-- @usage helpers.wait_timer("rate-limiting", true, "all-finish", 10)
```

This should work, but is incomplete. Can you check the `@return` tag and the description?

**Comment by ADD-SP** [comment]
> This doesn't even show up in the documentation? Can you try and run `ldoc` and verify?

截屏2022-07-21 18 29 02

**Comment by Tieske** [comment]
@ADD-SP yep,, my bad, it did show up. But is has the wrong annotations; '?' and '|' characters that are not recognized. My suggestion above takes care of that.

File: spec/helpers.lua [hunk 2]
```diff
@@ -1388,13 +1388,27 @@ local function wait_until(f, timeout, step)
...
+--- 
+---any-running: At least one of the timers that were matched is running
+--- 
+---all-running: All timers that were matched are running
+--- 
+---worker-wide-all-finish: All the timers in the worker that were matched finished
+---@param[opt=2] timeout number optional, maximum time to wait
+---@param[opt] admin_client_timeout number optional, to override the default timeout setting
+---@param[opt] forced_admin_port number optional, to override the default port of admin API
+---@usage helpers.wait_timer("rate-limiting", true, "all-finish", 10)
```
Comment (Tieske) [inline spec/helpers.lua:1411]: can we please make this consistent with the rest of the file? triple-dash + space on the first line, double-dash + space on the following lines

**Review by Tieske — CHANGES_REQUESTED** [review]

**Review by Tieske — APPROVED** [review]

**Pull request merged**